feature-libs/cart/main/core/facade/active-cart.service.ts

```diff
@@ -328,20 +284,18 @@ export class ActiveCartService implements ActiveCartFacade, OnDestroy {
     );
   }
 
-  requireLoadedCart(
-    customCartSelector$?: Observable<
-      StateUtils.ProcessesLoaderState<Cart | undefined>
-    >
-  ): Observable<StateUtils.ProcessesLoaderState<Cart | undefined>> {
+  requireLoadedCart(forGuestMerge = false): Observable<Cart> {
     // For guest cart merge we want to filter guest cart in the whole stream
     // We have to wait with load/create/addEntry after guest cart will be deleted.
-    // That's why you can provide custom selector with this filter applied.
-    const cartSelector$ = customCartSelector$
-      ? customCartSelector$
-      : this.cartSelector$;
+    const cartSelector$ = (
+      forGuestMerge
+        ? this.cartEntity$.pipe(
+            filter(() => !Boolean(getLastValueSync(this.isGuestCart())))
+          )
+        : this.cartEntity$
+    ).pipe(filter((cartState) => !cartState.loading));
 
     return cartSelector$.pipe(
-      filter((cartState) => !cartState.loading),
```

Comment:
Does it matter we don't check anymore if th cart is loading?

Reply:
It was added in the line 296.